### ISSUE TITLE
base memory consumption has increased since 1.10.0

### ISSUE THREAD

**Issue by garypen** [apollographql/router#3247]
1.10.0 is chosen as an arbitrary base in the past...

If we run the router with a variety of different supergraphs, then we can see that memory consumption has increased:

```
graph        1.10.0.       dev(ffda5f7e139f2ca0bdf091fe181f29a3dd6ce3da)
local         33            43
graph1      163           240
graph2      180           283
(measuring MB of memory consumed by the router just after startup, no requests serviced)
```
The absolute figures are still not very large, but this could have an impact in memory constrained environments and we should investigate what has caused this.
Note: These measurements were taken on MacOS, so we can probably eliminate `jemalloc` as a potential cause since we only enable that on Linux systems.
Note: The increase in memory consumption has also been observed on Linux systems.

**Comment by abernix** [comment]
@garypen Is this still the case? Should we take time to prioritize a bisection?

**Comment by garypen** [comment]
It is. We should.

**Comment by abernix** [comment]
It's not intuiative to me how to read the codeblock at the top of this issue.  It appears to be a table of sorts. 

That said, using the [starstuff supergraph](https://supergraph.demo.starstuff.dev/), base startup memory usage — which I measured after starting the router with the (starstuff) supergraph and making exactly one request to `{__typename}` — did gain marginally from 1.8.0 to 1.19.0, raising from about 8.3MB to 8.9MB (on Linux x64).

However, it them dropped by approximately half in Apollo Router 1.19.1 as of the landing of fbeaf57920e8f15073590f7dcd4cac548fb0e087, leaving it at about 4.5MB.

**Comment by garypen** [comment]
I did some bisecting:

```
branch memory (Mb)
dev      233
1.19.1   282
1.19.0   181
1.18.0   184
1.15.0   185
1.10.0   176
```
I looked at things which went into 1.19.1 and decided that the most likely looking cause was a router bridge update. I tried reverting that change, but found that it required `bytes=1.2.1`, so I also made that change. With both those changes in place

```
branch                                        memory (Mb)
1.19.1  - router_bridge - bytes                 189
```

Ok, that looks like a smoking gun. Something about either `bytes` or `router_bridge` update has led to a big increase in the amount of memory consumed by an executing router.  It's unlikely to be `bytes`. Looking at the router bridge update, which was from 0.2.4 to 0.2.5, and the only changes there were bumps to deno core functionality: from 171 to 187 on the main deno-core crate.

At this point, I don't think I can do much more. Looking into `deno` for reasons why memory consumption has increased is likely to be a time sink which generates nothing positive. I don't think reverting deno is a viable option (in fact, we are on deno-core 200 at this point in time).

Until we figure out a way to remove deno from the router, I think we just have to accept this change in memory footprint. I'll mark this as blocked until then.
